server.R

```diff
@@ -91,541 +92,580 @@ server <- function(input, output, session) {
 
   output$uiScreenButton <- renderUI({
     if (values$dataUploaded && values$metaUploaded) {
-      actionButton("screenbutton", "Screen files", width = "80%")
+      if (values$environment == "shinyapps") {
+        actionButton("external_screenbutton", "Screen files", width = "80%")
+      } else {
+        actionButton("screenbutton", "Screen files", width = "80%")
+      }
     }
   })
 
-  observeEvent(input$screenbutton, {
-    shinyjs::hideElement(id = "guidance")
-
-    # Show loading screen ---------------------------------------------------------------------------------
-    shinyjs::showElement(id = "loading", anim = TRUE, animType = "fade", time = 1)
-
-    values$shouldShow <- TRUE
-
-    # Disable buttons ---------------------------------------------------------------------------------
-
-    disable("uiScreenButton")
-    disable(selector = "span[class='btn btn-default btn-file']")
+  # Confirmation modal for external use ----------------------------------------------------------
+
+  observeEvent(input$external_screenbutton, {
+    shinyalert::shinyalert(
+      title = "",
+      text = "This is the external version of the DfE data screening app. We do not control the servers that this app runs on. You should only use this to screen published or dummy data. For sensitive files, you should clone the repository and screen files locally instead.",
+      type = "warning",
+      closeOnEsc = FALSE,
+      closeOnClickOutside = FALSE,
+      html = FALSE,
+      showConfirmButton = TRUE,
+      showCancelButton = TRUE,
+      confirmButtonText = "I understand, proceed with screening files",
+      confirmButtonCol = "#AEDEF4",
+      cancelButtonText = "Cancel screening",
+      animation = TRUE,
+      size = "s",
+      callbackR = function(x) {
+        if (x == TRUE) {
+          values$proceed_with_screening <- x
+        }
+      },
+      immediate = FALSE
+    )
+  })
 
-    # Read in files ---------------------------------------------------------------------------------
+  # Main screening button ------------------------------------------------------------------------
 
-    inputData <- values$datafile
-    inputMeta <- values$metafile
+  observeEvent(input$screenbutton | values$proceed_with_screening,
+    {
+      if (input$screenbutton == 0 && is.null(values$proceed_with_screening)) {
 
-    data <- readFile(inputData$datapath)
-    meta <- readFile(inputMeta$datapath)
+        # This ends it early
```

Comment:
This with the callback in the modal is so neat!! Could we also clear the inputData and inputMeta back to NULL in this step if it's not too faffy?

Reply:
Good thinking - I think I've managed to do this in the latest commit